📖 docs: updated flavors and machine pool docs (#2030)
kubernetes-sigs/cluster-api-provider-aws: docs/book/src/topics/machinepools.md

```

The template used for this [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors)
is located [here](../../templates/cluster-template-machinepool.yaml).

## AWSManagedMachinePool

> Cluster API Provider AWS (CAPA) has experimental support for [EKS Managed Node Groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) using `MachinePool` through the infrastructure type `AWSManagedMachinePool`. An `AWSManagedMachinePool` corresponds to an [AWS AutoScaling Groups](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html) that is used for an EKS managed node group. .
> 
> The AWSManagedMachinePool controller creates and manages an EKS managed node group with in turn manages an AWS AutoScaling Group of managed EC2 instance types.

Reviewer: ```suggestion
Cluster API Provider AWS (CAPA) has experimental support for [EKS Managed Node Groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) using `MachinePool` through the infrastructure type `AWSManagedMachinePool`. An `AWSManagedMachinePool` corresponds to such a managed node group which is in turn powered by an [AWS AutoScaling Group](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroup.html), that handles the orchestration of EC2 instances.
```

Reviewer: Seemed a little redundant as is

Author: Changed.

### Using `clusterctl` to deploy

> To deploy an EKS managed node group using AWSManagedMachinePool via `clusterctl config` you can use a [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors).

Reviewer: ```suggestion
To deploy an EKS managed node group using `AWSManagedMachinePool` via `clusterctl config` you can use a [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors).
```

Author: Changed.

> Make sure to set up your AWS environment as described [here](https://cluster-api.sigs.k8s.io/user/quick-start.html).

Reviewer 2: ```suggestion
Make sure to set up your AWS environment as described [here](../getting-started.md).
```

The getting started in this repo just includes the upstream CAPI doc.
